# Now, let's use our Random Forest Model to `Predict Harper's Apartment Listing Price`

In [14]:
!pip install forestci

## Entering Harper's Property Information and Predicting the Price of One Night: 

In [15]:
my_house = pd.DataFrame(columns = X_test.columns)
my_house.loc[len(my_house)] = 0
my_house['latitude'] = 40.7140890610426
my_house['longitude'] = -74.00906863228428
my_house['calculated_host_listings_count'] = 1
my_house['availability_365'] = 210
my_house['nearby_subway_line_counts'] = 12
my_house['Manhattan'] = 1
my_house['Tribeca'] = 1
my_house['Entire home/apt'] = 1
my_house
my_house_prediction = RF_airbnb.predict(my_house)[0]
my_house_prediction

5.9587921667649

### Now, let's convert the log_price prediction to display the acutal dollar amount: 

In [1]:
import math

In [3]:
normal_price = math.exp(my_house_prediction)

print(f"The normal price is approximately: {normal_price:.2f}")

The normal price is approximately: 387.14


### We get a baseline log_price prediction of `5.959`, which is approximately `387 dollars` per night. But how confident is this prediction?

### In order to check, we have to bootstrap our model as follows: 

In [16]:
#I am adding my house X values to the X_test dataframe
X_test.loc[len(X_test)] = my_house.loc[0]

In [20]:
import forestci as fci
import warnings
warnings.filterwarnings('ignore')
#forest SCI is a publicly available package that does exactly what we want
errors = fci.random_forest_error(RF_airbnb, X_train, X_test)

In [22]:
log_mh_min = my_house_prediction - errors[-1]
log_mh_max = my_house_prediction + errors[-1]
print("95% Confidence Interval for Log Price of My House Listing:")
print("2.5%", round(log_mh_min, 10))
print("50%", round(my_house_prediction, 10))
print("97.5%", round(log_mh_max, 10))

print()
mh_min = np.e**(log_mh_min)
mh_max = np.e**(log_mh_max)
print("95% Confidence Interval for Price of Test Listings:")
print("2.5%", round(mh_min, 10))
print("50%", round(np.e**my_house_prediction, 10))
print("97.5%", round(mh_max, 10))

95% Confidence Interval for Log Price of My House Listing:
2.5% 5.6627519401
50% 5.9587921668
97.5% 6.2548323934

95% Confidence Interval for Price of Test Listings:
2.5% 287.9399467446
50% 387.1422384683
97.5% 520.5221244942


### So, using our Random Forest model, we are 95% confident that the true price of Harper's listing will lie between `[$288 - $520]` based on our predictions. This range is relatively large, possibly due to the fact that Harper's listing is in a nice neighborhood - Tribeca - where there are fewer samples at this price point. 

### Now you can use this notebook to understand how much you should charge per night for `YOUR` property! 